### PR TITLE
Fix "sanityCheck" task

### PR DESCRIPTION
**Pull request by melix**
### Context

Without this change, running `sanityCheck` fails with the following error:

```
Could not determine the dependencies of task ':performance:checkstyleJmhGroovy'.
> Could not resolve all task dependencies for configuration ':performance:compile'.
   > Could not resolve project :distributionsDependencies.
     Required by:
         project :performance
      > Cannot choose between the following variants of project :distributionsDependencies:
          - apiElements
          - runtimeElements
        All of them match the consumer attributes:
          - Variant 'apiElements':
              - Found org.gradle.component.category 'platform' but wasn't required.
              - Found org.gradle.usage 'java-api' but wasn't required.
          - Variant 'runtimeElements':
              - Found org.gradle.component.category 'platform' but wasn't required.
              - Found org.gradle.usage 'java-runtime' but wasn't required.
```

It's unclear when this started to happen, but this is caused by the JMH plugin resolving the wrong configurations. A new version of the plugin has been released as part of fixing this.

### CODE REVIEW

**Comment by wolfs** [comment]
@melix We are running `sanityCheck` on CI as well, any idea why it succeeds there? https://e.grdev.net/s/fqlbsi4pdkkli/

**Comment by melix** [comment]
I don't, really. This error systematically happens locally: https://e.grdev.net/s/3sltv4zdqfono/failure?openFailures=WzBd&openStackTraces=WzNd#top=0

With this fix, it goes away.

File: buildSrc/subprojects/profiling/profiling.gradle.kts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 dependencies {
-    implementation("me.champeau.gradle:jmh-gradle-plugin:0.4.7")
+    implementation("me.champeau.gradle:jmh-gradle-plugin:0.4.8")
```
Comment (wolfs) [inline buildSrc/subprojects/profiling/profiling.gradle.kts]: Is the `jmh-gradle-plugin` update related to the failures?

Comment (melix) [reply buildSrc/subprojects/profiling/profiling.gradle.kts]: Yes, fixing this error required an upgrade.

**Comment by melix** [comment]
> The only downside I see is that this change blocks us from using plugins which for some reason resolve `compile` or `runtime`

This is intentional, and done to discover such plugins.

**Comment by wolfs** [comment]
@melix The change breaks IDEA sync for me: 
```
<ij_msg_gr>Project resolve errors<ij_msg_gr><ij_nav>/Users/wolfs/projects/gradle/build-tool/build.gradle.kts<ij_nav><i><b>root project 'gradle': Unable to resolve additional project configuration.</b><eol>Details: java.lang.IllegalStateException: Resolving configuration 'compile' directly is not allowed</i>
<ij_msg_gr>Project resolve errors<ij_msg_gr><ij_nav>/Users/wolfs/projects/gradle/build-tool/build.gradle.kts<ij_nav><i><b>root project 'gradle': Unable to resolve additional project configuration.</b><eol>Details: java.lang.IllegalStateException: Resolving configuration 'compile' directly is not allowed</i>
<ij_msg_gr>Project resolve errors<ij_msg_gr><ij_nav>/Users/wolfs/projects/gradle/build-tool/subprojects/announce/announce.gradle.kts<ij_nav><i><b>project ':announce': Unable to resolve additional project configuration.</b><eol>Details: java.lang.IllegalStateException: Resolving configuration 'compile' directly is not allowed</i>
<ij_msg_gr>Project resolve errors<ij_msg_gr><ij_nav>/Users/wolfs/projects/gradle/build-tool/subprojects/antlr/antlr.gradle.kts<ij_nav><i><b>project ':antlr': Unable to resolve additional project configuration.</b><eol>Details: java.lang.IllegalStateException: Resolving configuration 'compile' directly is not allowed</i>
<ij_msg_gr>Project resolve errors<ij_msg_gr><ij_nav>/Users/wolfs/projects/gradle/build-tool/subprojects/architecture-test/architecture-test.gradle.kts<ij_nav><i><b>project ':architectureTest': Unable to resolve additional project configuration.</b><eol>Details: java.lang.IllegalStateException: Resolving configuration 'compile' directly is not allowed</i>
```

I'll revert it on `master`.

**Comment by wolfs** [comment]
Does it help if we mark the configurations only as `canBeConsumed = false`?

**Comment by melix** [comment]
It would, but interest is lower.

**Comment by wolfs** [comment]
@melix Do you think you can open an issue on the Jetbrains Youtrack to make them aware of the possible problem?

**Pull request merged**